From 5ca1443dab1342901b507adfeeea0fd1f2605e79 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Fri, 17 Dec 2010 11:07:23 +0800
Subject: [PATCH] ENGR00137101-2 NAND : change the field `is_onfi_nand` to `is_ddr_ok`

The ONFI NAND and TOGGLE NAND both support the DDR.
So merge the same attribution to a new field `is_ddr_ok`.

Also add a inline function to judge the DDR nand.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 drivers/mtd/nand/nand_device_info.h |   15 +++++++++------
 1 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/drivers/mtd/nand/nand_device_info.h b/drivers/mtd/nand/nand_device_info.h
index 63011c7..f74c975 100644
--- a/drivers/mtd/nand/nand_device_info.h
+++ b/drivers/mtd/nand/nand_device_info.h
@@ -43,7 +43,7 @@ enum nand_device_cell_technology {
  *                             the device.
  * @device_code:               The device code (2nd ID byte) reported by the
  *                             device.
- * @is_onfi_nand:              Is this nand an ONFI nand ?
+ * @is_ddr_ok:                 Is this nand an ONFI nand or TOGGLE nand ?
  * @cell_technology:           The storage cell technology.
  * @chip_size_in_bytes:        The total size of the storage behind a single
  *                             chip select, in bytes. Notice that this is *not*
@@ -90,8 +90,8 @@ struct nand_device_info {
 	uint8_t   manufacturer_code;
 	uint8_t   device_code;
 
-	/* ONFI nand */
-	bool	  is_onfi_nand;
+	/* Does the nand support DDR? (ONFI or TOGGLE) */
+	bool	  is_ddr_ok;
 
 	/* Technology */
 
@@ -142,8 +142,11 @@ struct nand_device_info *nand_device_get_info(const uint8_t id_bytes[]);
 void nand_device_print_info(struct nand_device_info *info);
 
 /*
- * Check the NAND whether it is an ONFI nand.
+ * Check the NAND whether it supports the DDR mode.
+ * Only the ONFI nand and TOGGLE nand support the DDR now.
  */
-bool is_onfi_nand(struct nand_device_info *info);
-
+static inline bool is_ddr_nand(struct nand_device_info *info)
+{
+	return info->is_ddr_ok;
+}
 #endif
-- 
1.5.4.4

